colossalai/shardformer/policies/basepolicy.py

```diff
@@ -99,137 +74,75 @@ class for the example.
 
     """
 
-    @staticmethod
-    def argument_policy(model_config, world_size: int) -> Dict[nn.Module, Argument]:
+    def __init__(self) -> None:
+        self.model = None
+
+    def set_model(self, model: nn.Module) -> None:
         r"""
-        Return the dict for the modify policy, the key is the original layer class and the value is the
-        argument for the modify layer
+        Set model as an attribute of the Policy object so that we can access the model's attributes.
 
         Args:
-            model_config (:class:`tansformer.Config`): The config of transformer model
-            world_size (int)): The world size of sharding model
+            model (:class:`nn.Module`): The model to be perform
+        """
+        self.model = model
+
+    @abstractmethod
+    def preprocess(self, shard_config: ShardConfig = None) -> nn.Module:
+        r"""
+        Perform some preprocessing of the model, like reshaping the embedding layer
+        """
+
+    @abstractmethod
+    def module_policy(self, shard_config: ShardConfig = None) -> Dict[Union[str, nn.Module], ModulePolicyDescription]:
+        r"""
+        Return the dict for the modify policy, the key is the original layer class and the value is the
+        argument for the modify layer
 
         Return:
             Dict for the modify policy,
             ::
             {
-                origin layer class1 (nn.Module): Argument(
-                    attr_dict = {
-                        argument1: value1,
-                        argument2: value2,
+                origin layer class1 (nn.Module): ModulePolicyDescription(
+                    attribute_replacement = {
+                        "attribute1": value1,
+                        "attribute2": value2,
                         ...
                     },
-                    param_funcs = [
+                    param_replacement = [
                         staticmethod1,
                         staticmethod2,
                         ...
-                    ]
-                ),
-                origin layer class2 (nn.Module): Argument(
-                    attr_dict = {
-                        argument1: value1,
-                        argument2: value2,
-                        ...
-                    },
-                    param_funcs = [
+                    ],
+                    sub_module_replacement = [
```

Comment:
The element in the `sub_module_replacement` list is not staticmethod, but `SubModuleReplacementDescription`.